crates/re_viewer/src/ui/selection_panel.rs

```diff
@@ -413,6 +408,226 @@ fn entity_props_ui(
         });
 }
 
+fn visible_history_ui(
+    ctx: &mut ViewerContext<'_>,
+    ui: &mut Ui,
+    space_view_class_name: &SpaceViewClassName,
+    entity_path: Option<&EntityPath>,
+    entity_props: &mut EntityProperties,
+) {
+    //TODO(#4107): support more space view types.
+    if space_view_class_name != "3D" && space_view_class_name != "2D" {
```

Comment:
since we're in re_viewer you should be able to use the symbolic names - `re_space_view_spatial::SpatialSpaceView2D::name()` etc.

then again it would be great to *not* be in re_viewer (we want to keep it as small as possible; it used to be that pretty much everything was in here, now it's the root node of a loot of dependencies) and in that case you might need this workaround (and moving this out is out of scope of this PR)

... so no strong preference

Reply:
I considered it but sadly `::name()` takes `&self`, not `Self`. Unless I miss something, we should extract a `pub const` for those names and use them in `name()`'s implementation?

Reply:
ah you're right. They can't be Self taking because of trait shenanigans (these ones must be "object safe" otherwise the downcast to Any doesn't work)

Reply:
I did add a `const` in 5ddf00acfe33ea4d1fe829872cfd87a15bf6c1f3, which is a tad cleaner imo.